libbeat/processors/ratelimit/rate_limit.go

```diff
@@ -40,7 +41,10 @@ const processorName = "rate_limit"
 type rateLimit struct {
 	config    config
 	algorithm algorithm
-	logger    *logp.Logger
+
+	numRateLimited atomic.Uint64
```

Comment:
I decided to keep track of the number of events that have been rate-limited (since the last time an event was allowed through; more on that in https://github.com/elastic/beats/pull/23330/files#r550323522). Alternatively we could simply have a boolean to indicate that rate limiting has happened. But I felt like a number might be more informative. Happy to discuss and change, though!

Reply:
On a related note, it would be nice if we could include this metric as part of the libbeat monitoring infrastructure. For example, the `dns` processor is doing something like this. If you agree, I can implement this as part of this PR or a separate PR.